lightning/src/ln/channelmanager.rs

```diff
@@ -7067,6 +7078,139 @@ where
 		}
 	}
 
+	/// Creates an [`OfferBuilder`] such that the [`Offer`] it builds is recognized by the
+	/// [`ChannelManager`] when handling [`InvoiceRequest`] messages for the offer.
+	///
+	/// Uses a one-hop [`BlindedPath`] for the offer with [`ChannelManager::get_our_node_id`] as the
+	/// introduction node and a derived signing pubkey for recipient privacy. As such, currently,
+	/// the node must be announced. Otherwise, there is no way to find a path to the introduction
+	/// node in order to send the [`InvoiceRequest`].
+	///
+	/// [`Offer`]: crate::offers::offer::Offer
+	/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+	pub fn create_offer_builder(
+		&self, description: String
+	) -> OfferBuilder<DerivedMetadata, secp256k1::All> {
```

Comment:
May want to set a sane default for the expiry. It may also be nice in the future to add accessor methods on `OfferBuilder` so users can tell what we've set.

Reply:
BOLT 11 invoices default to 1 hour expiry if none is specified, so it may be confusing that there's a different default behavior here. A comment may be useful but fine if you don't think it'll be confusing.

Reply:
Yeah, BOLT 12 invoices also have a default expiry, but offers do not. I'd prefer to at least not do anything to surprising with regard to the spec, at least. Let me update the docs.